startup.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+import re
+import glob
+
+from sgtk.platform import SoftwareLauncher, SoftwareVersion, LaunchInformation
+
+class PhotoshopLauncher(SoftwareLauncher):
+    """
+    Handles the launching of Photoshop. Contains the logic for
+    scanning for installed versions of Photoshop and
+    how to correctly set up a launch environment for the photoshop
+    engine.
+    """
+
+    def scan_software(self, versions=None):
+        """
+        Performs a scan for software installations.
+
+        :param list versions: List of strings representing versions
+                              to search for. If set to None, search
+                              for all versions. A version string is
+                              DCC-specific but could be something
+                              like "2017", "6.3v7" or "1.2.3.52".
+        :returns: List of :class:`SoftwareVersion` instances
+        """
+        icon_path = os.path.join(self.disk_location, "resources", "ps_2017_icon_256.png")
```

Comment:
the app icon is picked up from the engine (rather than messing with trying to extract it from the executable).